tests/Unit/NumericalAlgorithms/Spectral/Test_ChebyshevGauss.cpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace Spectral {
+// To test the \f$w_k\f$ as opposed to the Spectral::quadrature_weights. We
+// don't provide a public accessor for the former since we only ever need them
+// in Spectral.cpp.
+template <Basis, Quadrature>
+std::pair<DataVector, DataVector> compute_collocation_points_and_weights(
+    size_t) noexcept;
+}  // namespace Spectral
+
+namespace {
+
+void test_points_and_weights(const size_t num_points,
+                             const DataVector expected_points,
+                             const DataVector expected_weights) {
+  const auto& points =
+      Spectral::collocation_points<Spectral::Basis::Chebyshev,
+                                   Spectral::Quadrature::Gauss>(num_points);
+  CHECK_ITERABLE_APPROX(expected_points, points);
+  // We test the \f$w_k\f$ here directly. Test_Spectral.cpp and
```

Comment:
Is it necessary to test a private function?  Presumably something else would fail if it was implemented incorrectly...

Reply:
Yes `definite_integral` should fail, and a lot of other tests too. This tests that the algorithm to compute points and weights works as expected, which would be hard to debug otherwise.